data/beaconrestapi/src/integration-test/java/tech/pegasys/teku/beaconrestapi/AbstractDataBackedRestAPIIntegrationTest.java

```diff
@@ -109,10 +112,13 @@ private void setupStorage(final StorageSystem storageSystem, final boolean useMo
     recentChainData = storageSystem.recentChainData();
     chainBuilder = ChainBuilder.create(VALIDATOR_KEYS);
     chainUpdater = new ChainUpdater(recentChainData, chainBuilder);
-    this.forkChoice =
-        useMockForkChoice
-            ? mock(ForkChoice.class)
-            : new ForkChoice(recentChainData, stateTransition);
+    if (useMockForkChoice) {
+      beaconChainUtil = mock(BeaconChainUtil.class);
```

Comment:
Mocking `BeaconChainUtil` is unexpected given its a test util class. I think I'd just always use the real `BeaconChainUtil`. The only place it's used is for the non-mock version anyway.

Reply:
its only while mocking the fork choice so that tests pass, otherwise the 'not ready' tests fail... i can investigate if there's another way to do this...

Reply:
They passed when I ran them with the change.  Which ones were failing for you?

Reply:
There's only a couple of tests that ever call the method that uses `beaconChainUtil` - the others should be unaffected (we'd still mock `FockChoice` just not `BeaconChainUtil`.